feature/gnmi/otg_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -502,19 +503,20 @@ func TestSoftwareVersion(t *testing.T) {
 		if v, ok := parent.Val(); ok {
 			got := gnmi.Get(t, dut, gnmi.OC().Component(v).Type().State())
 
-			// Arista_7280 OC component EOS has parent type Chassis
-			if dut.Model() == "DCS-7280CR3K-32D4" {
-				if got == chassisType {
-					t.Logf("Got a valid parent %v with a type %v for the component %v", v, got, os)
-				} else {
-					t.Errorf("Got a parent %v with a type %v for the component %v, want %v", v, got, os, chassisType)
-				}
+			want := []oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT{supervisorType}
+			if deviations.OSComponentParentIsChassis(dut) {
+				want = []oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT{chassisType}
+			}
+			if deviations.OSComponentParentIsSupervisorOrLinecard(dut) {
+				want = []oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT{supervisorType, linecardType}
+			}
+
+			if slices.IndexFunc(want, func(w oc.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) bool {
+				return w == got
+			}) != -1 {
```

Comment:
It's easier to read conditions with a positive polarity, i.e. I would change this to check "== -1" and swap the Errorf and Logf statements.

Reply:
Got it. Fixed.